pages/index.tsx

```diff
@@ -64,31 +82,22 @@ export async function getStaticProps() {
 
   return {
     props: {
-      sections: response.sectionCollection.items,
+      sections: response?.sectionCollection?.items ?? [],
     },
   }
 }
 
 /**
- * Map content types to components and fields to props
+ * Map content types to components using adapters
  *
  */
 function renderPresenter(type: ContentType, fields: any): React.ReactElement {
-  const id = camelCase(fields.title)
-
   const componentMap = {
-    Hero: (
-      <Hero id={id} title={fields.title} description={fields.heroDescription} />
-    ),
-    ContentBlock: (
-      <ContentPanel id={id}>
-        <LeftCol>
-          {fields.title && <h2>{fields.title}</h2>}
-          {fields?.description?.json &&
-            documentToReactComponents(fields.description.json)}
-        </LeftCol>
-      </ContentPanel>
-    ),
+    Hero: <HeroAdapter fields={fields} />,
```

Comment:
Again here I wonder if we can use `[Component.name]`.. just trying to avoid having multiple points of maintenance. Maybe there is something more dynamic that can be done so that we don't need to update multiple places when we add a new component + adapter?

Reply:
Problem with this is they become less meaningful to an end user in Contentful. 

`isRequired` works in code but not in the CMS